src/main/scala/org/ergoplatform/explorer/persistence/repositories/HeaderRepo.scala

```diff
@@ -0,0 +1,57 @@
+package org.ergoplatform.explorer.persistence.repositories
+
+import cats.effect.Sync
+import cats.implicits._
+import doobie.util.transactor.Transactor
+import doobie.implicits._
+import doobie.refined.implicits._
+import org.ergoplatform.explorer.Id
+import org.ergoplatform.explorer.persistence.models.Header
+
+/** [[Header]] data access operations.
+  */
+trait HeaderRepo[F[_]] {
+
+  /** Put a given `h` to persistence.
+    */
+  def insert(h: Header): F[Unit]
+
+  /** Update a header with a given `h.id` according to a new values from `h`.
+    */
+  def update(h: Header): F[Unit]
+
+  /** Get header with a given `id`.
+    */
+  def get(id: Id): F[Option[Header]]
+
+  /** Get all headers at the given `height`.
```

Comment:
Only one header at the given height, no?

Reply:
Forks are possible